providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/EventStreamObserver.java

```diff
@@ -0,0 +1,74 @@
+package dev.openfeature.contrib.providers.flagd;
+
+import java.util.Map;
+import io.grpc.stub.StreamObserver;
+import lombok.extern.slf4j.Slf4j;
+import dev.openfeature.flagd.grpc.Schema.EventStreamResponse;
+import com.google.protobuf.Value;
+
+/**
+ * EventStreamObserver handles events emitted by flagd.
+ */
+@Slf4j
+public class EventStreamObserver implements StreamObserver<EventStreamResponse> {
+    private EventStreamCallback callback;
+    private FlagdCache cache;
+
+    private static final String configurationChange = "configuration_change";
+    private static final String providerReady = "provider_ready";
+
+    EventStreamObserver(FlagdCache cache, EventStreamCallback callback) {
+        this.cache = cache;
+        this.callback = callback;
+    }
+
+    @Override
+    public void onNext(EventStreamResponse value) {
+        switch (value.getType()) {
+            case configurationChange:
+                this.handleConfigurationChangeEvent(value);
+                break;
+            case providerReady:
+                this.handleProviderReadyEvent();
+                break;
+            default:
+                log.warn("unhandled event type {}", value.getType());
```

Comment:
```suggestion
                log.debug("unhandled event type {}", value.getType());
```

We actually get a lot of other events ("keep-alive") for instance. I would recommend we make this a `debug` to avoid noise.